cli/dcoscli/node/main.py

```diff
@@ -65,12 +67,324 @@ def _cmds():
             function=_ssh),
 
         cmds.Command(
+            hierarchy=['node', 'snapshot', 'create'],
+            arg_keys=['<nodes>'],
+            function=_snapshot_create),
+
+        cmds.Command(
+            hierarchy=['node', 'snapshot', 'delete'],
+            arg_keys=['<snapshot>'],
+            function=_snapshot_delete),
+
+        cmds.Command(
+            hierarchy=['node', 'snapshot', 'download'],
+            arg_keys=['<snapshot>', '--to'],
+            function=_snapshot_download),
+
+        cmds.Command(
+            hierarchy=['node', 'snapshot'],
+            arg_keys=['--list', '--status', '--cancel', '--json'],
+            function=_snapshot_manage),
+
+        cmds.Command(
             hierarchy=['node'],
             arg_keys=['--json'],
-            function=_list),
+            function=_list)
     ]
 
 
+def snapshot_error(fn):
+    @functools.wraps(fn)
+    def check_for_snapshot_error(*args, **kwargs):
+        response = fn(*args, **kwargs)
+        if response.status_code != 200:
+            err_msg = ('Error making {} request\nURL: '
+                       '{}, status_code: {}.'.format(args[1], args[0],
```

Comment:
we allow 503 and 400 status codes, because they also bring the error message back to user